### ISSUE TITLE
Crash in RustLogins/Places forceClose

### ISSUE THREAD

**Issue by garvankeeley** [mozilla-mobile/firefox-ios#5597]
This is our second-to-top crasher
<img width="962" alt="image" src="https://user-images.githubusercontent.com/5495083/66666152-09da8280-ec1e-11e9-833f-0a575ccac0af.png">

**Comment by garvankeeley** [comment]
iOS 13 increased the incidence of this:
<img width="242" alt="image" src="https://user-images.githubusercontent.com/5495083/66666185-21b20680-ec1e-11e9-8d23-848a233b4c9f.png">

**Comment by athomasmoz** [comment]
Blocked on android services https://github.com/mozilla/application-services/issues/1964

**Comment by garvankeeley** [comment]
Severity: it is our 2nd to top crasher, it would be nice to have it addressed sooner rather than later.
User experience: the app is killed when it is in background, one example of impact is the user will lose state of web pages if they were in the middle of doing something when they backgrounded.

**Comment by garvankeeley** [comment]
The problem is that we perform a sync when the app is backgrounded, it takes >30s, we then try to close the db in `expirationHandler` with a `forceClose`, this blocks the main thread. Blocking the main thread causes the app to be killed.
I experimented with having busy background threads when the app is in the background, and this does not cause a crash. At the time the background task expires, the OS pauses thread execution, and resumes it once the app is foregrounded.
Conclusions:
1) We should only close the DB after a sync is completed
2) If a sync is not completed, don't try close the db and block
3) It is fine to leave the sync bg task running at the time of bg task expiration
